### ISSUE TITLE
Images without true objects considered as negative sample in hard negative mining 

### ISSUE THREAD

**Issue by nyanmn** [weiliu89/caffe#909]
If I include images without true objects, would it be considered as negative samples and put into hard negative mining like in Yolo.
If I like to include objects purposely as negative samples to do negative mining, what should I do in SSD?

**Comment by fraukej** [comment]
It seems that @weiliu89 didn't implement it. You could do it yourself.
See #146 and #612 for more information.

**Comment by nyanmn** [comment]
Hi,
I make a few images with all false positive detected objects as background and I put true positive object on top. 
Then train. Since positive to negative ration is 1:3, they should be taken in hard negative mining.
Does it make sense?

**Comment by fraukej** [comment]
@nyanmn 
Just to make sure I understood you correctly:
You take an example of an image that's easily considered an object when it's not (false positive) and which contains no ground truth objects. Then you photoshop a ground truth object onto that image to obtain a training example.

Yes, that makes sense, but I dont' think using that method yields the same results as using an image without any true objects for training.

May I ask how well trained your network is, e.g. number of iterations and batch size? I can image using an underfitted network to cause this problem.

**Comment by nyanmn** [comment]
Yes I use Paint (in Windows PC). I am still training now. Once I completed, I'll show loss, accuracy, mAP with charts. So that we can discuss with some results. The idea is different from Yolo approach. I could reject some false positive detection using that approach.

**Comment by nyanmn** [comment]
@fraukej @weiliu89 Actually it is same right? Doing hard negative mining from an image without true positive image and hard negative mining from an image with true positive object & bounding box in which the whole background is prepared with different sizes of false positive objects to be rejected. What is the difference?

**Comment by fraukej** [comment]
Yes, thank you! I'm looking forward to your results.

On your second comment: You might be right considering objects which don't intersect each other. But in my case I have false positives which span across the entire image (more or less). Adding a ground truth object on top of that yield different results, I think.

**Comment by nyanmn** [comment]
@fraukej Yeah yours is different issue. Mine is I am getting false positive detections consistently from same objects.
If they are consistently from same objects, then can prepare such images for different object which causes false positive detection

**Comment by fraukej** [comment]
@nyanmn What do you think, would it be possible to add an image to that dataset without any labelled boxes (that means an `image.jpg` and a corresponding `annotation.xml/txt` which is empty) and then calling `create_list` and `create_data`. Would it be possible to create a negative sample this way?

**Comment by nyanmn** [comment]
@fraukej yes that is another way. but hard negative mining algo part in SSD needs modification. Once that modification is done, it is one way of creating negative samples. Please share once you are done :)

**Comment by nyanmn** [comment]
@fraukej but please make sure true positives are not in that image.
